components/Charts/PieChart.js

```diff
@@ -0,0 +1,34 @@
+import { Chart as ChartJS } from "chart.js/auto";
+import { Pie } from "react-chartjs-2";
+import useStore from "../../store/useStore";
+
+export default function PieChart() {
+  const gameList = useStore((state) => state.games);
+  // Calculate won/lost games
+
+  let gamesWon = 0;
+  let gamesLost = 0;
+  gameList.map((game) => {
+    if (game.results.gameresult === "won") {
+      gamesWon = ++gamesWon;
+    }
+    if (game.results.gameresult === "lost") {
+      gamesLost = ++gamesLost;
+    }
+  });
```

Comment:
You are using the map function for a side effect. You should use forEach for side effects. You can also use reduce to return an object with gamesWon and gamesLost peroperties.